backend/apps/volontulo/tests/views/api/test_contact.py

```diff
@@ -0,0 +1,158 @@
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from apps.volontulo.factories import UserFactory
+from apps.volontulo.serializers import ContactSerializer
+
+ENDPOINT_URL = reverse('contact')
+
+
+def get_admin_email():
+    user = get_user_model().objects.filter(
+        userprofile__is_administrator=True,
+    ).first()
+    if user is None:
+        user = UserFactory.create()
+        user.userprofile.is_administrator = True
+        user.userprofile.save()
+    return user.email
+
+
+class TestContactViewGET(TransactionTestCase):
+
+    def test_get(self):
+        users = [UserFactory.create() for _ in range(5)]
+
+        users[0].email = 'abc@abc.pl'
+        users[0].userprofile.is_administrator = True
+        users[0].userprofile.save()
+        users[0].save()
+
+        users[1].email = 'xyz@xyz.com'
+        users[1].userprofile.is_administrator = True
+        users[1].userprofile.save()
+        users[1].save()
+
+        res = self.client.get(ENDPOINT_URL)
+
+        self.assertDictEqual(res.data, {
+            'administrator_emails': [users[0].email, users[1].email],
+            'applicant_types': ContactSerializer.APPLICANT_CHOICES,
+        })
+
+
+class TestContactViewPOST(APITestCase, TransactionTestCase):
+
+    def setUp(self):
+        self.payload = {
+            'administrator_email': get_admin_email(),
+            'applicant_email': 'michal.kowalski@wp.pl',
+            'applicant_name': 'Michał Kowalski',
+            'applicant_type': ContactSerializer.VOLUNTEER,
+            'message': 'Chcę zaangażować się w jakąś akcję, ale nie wiem jak.'
+                       ' Możecie mi pomóc :)?',
+            'phone_no': '+ 00 48 600 700 800',
+        }
+
+    def test_missing_fields(self):
+        res = self.client.post(ENDPOINT_URL, {}, format='json')
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(res.data, {
+            'administrator_email': ['To pole jest wymagane.'],
+            'applicant_email': ['To pole jest wymagane.'],
+            'applicant_name': ['To pole jest wymagane.'],
+            'applicant_type': ['To pole jest wymagane.'],
+            'message': ['To pole jest wymagane.'],
+            'phone_no': ['To pole jest wymagane.'],
+        })
+
+    def test_min_length(self):
+        self.payload.update({
+            'applicant_name': 'K',
+            'message': 'Cześć!'
+        })
+        res = self.client.post(ENDPOINT_URL, self.payload, format='json')
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertDictEqual(res.data, {
+            'applicant_name': [
+                'Upewnij się, że pole ma co najmniej 3 znaków.',
+            ],
+            'message': [
+                'Upewnij się, że pole ma co najmniej 10 znaków.',
+            ],
+        })
+
+    def test_max_length(self):
+        self.payload.update({
```

Comment:
I'm not a fan of setting `self.payload` in `setUp` and then doing `self.payload.update` in tests (I don't know if it will always work, as AFAIK nobody ensure, that tests will be run in particular order.